File: app/src/main/java/strip/PhotosUtils.java
package strip;


import android.app.Activity;
import android.content.ContentValues;
import android.content.Context;
import android.content.Intent;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.net.Uri;
import android.os.Environment;
import android.provider.MediaStore;
import android.widget.ImageView;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

public class  PhotosUtils {
    private static File writeImageView(ImageView imageView) {
        BitmapDrawable draw = (BitmapDrawable) imageView.getDrawable();
        Bitmap bitmap = draw.getBitmap();

        FileOutputStream outStream;
        File sdCard = Environment.getExternalStorageDirectory();
        File dir = new File(sdCard.getAbsolutePath() + "/commitStrip");
        dir.mkdirs();
        String fileName = String.format("%d.jpg", System.currentTimeMillis());
        File outFile = new File(dir, fileName);
        try {
            outStream = new FileOutputStream(outFile);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
            return null;
        }
        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, outStream);
        try {
            outStream.flush();
            outStream.close();
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
        return outFile;
    }

    public static boolean save(Activity activity, ImageView imageView) {
        File f = writeImageView(imageView);
        if (f == null)
            return false;
        Uri u = Uri.fromFile(f);
        Intent intent = new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE);
        intent.setData(u);
        activity.getBaseContext().sendBroadcast(intent);
        return true;
    }


    private static Uri getImageContentUri(Context context, String absPath) {

        Cursor cursor = context.getContentResolver().query(
                MediaStore.Images.Media.EXTERNAL_CONTENT_URI
                , new String[]{MediaStore.Images.Media._ID}
                , MediaStore.Images.Media.DATA + "=? "
                , new String[]{absPath}, null);

        if (cursor != null && cursor.moveToFirst()) {
            int id = cursor.getInt(cursor.getColumnIndex(MediaStore.MediaColumns._ID));
            cursor.close();
            return Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, Integer.toString(id));

        } else if (!absPath.isEmpty()) {
            ContentValues values = new ContentValues();
            values.put(MediaStore.Images.Media.DATA, absPath);
            cursor.close();
            return context.getContentResolver().insert(
                    MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values);
        } else {
            cursor.close();
            return null;
        }
    }
}

File: app/src/main/java/strip/PhotosUtils.kt
package strip


import android.content.Intent
import android.net.Uri
import android.os.AsyncTask
import android.os.Environment
import app.MainActivity
import app.R
import org.apache.commons.io.FileUtils
import java.io.File
import java.net.URL
import java.util.regex.Pattern

class PhotosUtils : AsyncTask<Any, Void, Boolean>() {
    override fun doInBackground(vararg p: Any?): Boolean {
        val activity = p[0] as MainActivity?
        val url = p[1] as String?

        if (activity != null && url != null) {
            try {
                val f = writeImage(url) ?: return false
                val u = Uri.fromFile(f)
                val intent = Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE)
                intent.data = u
                activity.baseContext.sendBroadcast(intent)
                activity.popUpDown(activity.getString(R.string.saveSuccess))
            } catch(e: Exception) {
                e.printStackTrace()
                activity.popUpDown(activity.getString(R.string.saveFailed))
            }

            return true
        }

        return false
    }

    private fun writeImage(url: String): File? {
        val sdCard = Environment.getExternalStorageDirectory()
        val dir = File(sdCard.absolutePath + "/commitStrip")
        dir.mkdirs()
        var extension = ""
        val p = Pattern.compile("^.*\\.([A-Za-z]{3,4})$")
        val m = p.matcher(url)
        if (m.matches() && m.groupCount() == 1) {
            extension = m.group(1)
        } else {
            throw Exception("L'url ne correspond pas au pattern : " + url)
        }
        val fileName = String.format("%d.$extension", System.currentTimeMillis())
        val outFile = File(dir, fileName)
        FileUtils.copyURLToFile(URL(url), outFile)

        return outFile
    }
}
